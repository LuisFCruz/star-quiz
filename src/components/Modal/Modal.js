import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Dialog, DialogTitle, withStyles, DialogActions, Button, DialogContent } from '@material-ui/core';
import Avatar from '../Avatar/Avatar';

const styles = theme => ({
  content: {
    display: 'flex',
    flexDirection: 'column',
    [theme.breakpoints.up('sm')]: {
      flexDirection: 'row',
    },
  },
  media:{
    maxWidth: '200px',
    maxHeight: '250px'
  },
  details: {
    paddingLeft: '10px',
  },
  label: {
    fontWeight: '700',
  }
});

export class Modal extends Component {
  static propTypes = {
    onClose: PropTypes.func.isRequired,
    open: PropTypes.bool.isRequired,
    character: PropTypes.object
  }

  handleClose = () => {
    this.props.onClose();
  };

  render() {
    const { character, open, classes } = this.props;

    let details = '';

    if (character) {
      const {
        id,
        height,
        hair_color,
        films,
        species,
        homeworld,
        vehicles
      } = character;
      details = (
        <DialogContent className={ classes.content }>
          <Avatar id={id} className={ classes.media } />
          <div className={ classes.details }>
            <p>
              <span className={classes.label}>Specie:</span> {species}
            </p>
            <p>
              <span className={classes.label}>Height:</span> {height}
            </p>
            <p>
              <span className={classes.label}>Hair:</span> {hair_color}
            </p>
            <p>
              <span className={classes.label}>Planet:</span> {homeworld}
            </p>
            <p>
              <span className={classes.label}>Movies:</span> {films}
            </p>
            <p>
              <span className={classes.label}>Vehicles:</span> {vehicles}
            </p>
          </div>
        </DialogContent>
      );
    }

    return (
      <Dialog onClose={this.handleClose} open={open}>
        <DialogTitle id="simple-dialog-title">Details</DialogTitle>
        {details}
        <DialogActions>
          <Button onClick={this.handleClose} color="primary">
            Close
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

export default withStyles(styles)(Modal);
