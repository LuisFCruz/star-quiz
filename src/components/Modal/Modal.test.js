import { shallow } from 'enzyme';
import React from 'react';

import { Modal } from './Modal';
import { Button } from '@material-ui/core';


describe('<Modal />', () => {
  let wrapper;
  let component;
  const character = {
      id: '1',
      height: '178',
      hairColor: 'red',
      films: 'star wars',
      species: 'human',
      homeworld: 'earth',
      vehicles: 'car',
  };
  beforeEach(() => {
    const props = {
      character,
      classes: {},
      selectCharacter: jest.fn()
    };
    wrapper = shallow(<Modal { ...props } />);
    component = wrapper.instance();
  });

  test('render with character', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render without character', () => {
    const props = { character: null, classes: {} };
    const localWrapper = shallow(<Modal { ...props } />);
    expect(localWrapper).toMatchSnapshot();
  });

  test('should have call handleClose()', () => {
    const handleSpy = jest.fn();
    component.handleClose = handleSpy;
    component.forceUpdate();

    const button = wrapper.find(Button);
    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(handleSpy).toHaveBeenCalled();
  });
  
  test('should have call selectCharacter()', () => {
    component.handleClose();
    expect(component.props.selectCharacter).toHaveBeenCalledTimes(1);
  });
});
