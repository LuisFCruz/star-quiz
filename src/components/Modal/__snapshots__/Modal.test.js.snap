// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> render with character 1`] = `
<WithStyles(Dialog)
  onClose={[Function]}
  open={true}
>
  <WithStyles(DialogTitle)>
    Details
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <Avatar
      id="1"
    />
    <div>
      <p>
        <span>
          Specie:
        </span>
         
        human
      </p>
      <p>
        <span>
          Height:
        </span>
         
        178
      </p>
      <p>
        <span>
          Hair:
        </span>
         
        red
      </p>
      <p>
        <span>
          Planet:
        </span>
         
        earth
      </p>
      <p>
        <span>
          Movies:
        </span>
         
        star wars
      </p>
      <p>
        <span>
          Vehicles:
        </span>
         
        car
      </p>
    </div>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Close
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`<Modal /> render without character 1`] = `""`;
